### PR TITLE
Implemented user creation with password salting and hashing

### PR DESCRIPTION
**Pull request by SamJDriver**
- Updated endpoint to take in limited json rather the whole
- Implemented password and salting hashing
- Implemented creation of new users

### CODE REVIEW

File: GameReview/BusinessLogic/Infrastructure/UserService.cs [hunk 1]
```diff
@@ -20,11 +17,46 @@ public UserService(GenericRepository<NickDbContext> genericRepository)
...
+
+            var existingUsernameUser = _genericRepository.GetSingleNoTrack<Users>(u => u.Username == user.Username);
+            if (existingUsernameUser != default)
+            {
+                error = "Username already exists. Please try a different username";
+                return -1;
+            }
+
+            byte[] salt = RandomNumberGenerator.GetBytes(128 / 8); // divide by 8 to convert bits to bytes
+            user.PasswordHash = saltPassword(user.PasswordHash, salt);
```
Comment (nisule) [inline GameReview/BusinessLogic/Infrastructure/UserService.cs]: I think we need to rename this user.PasswordHash to user.Password cuz this variable isn't the hash yet it is the plaintext password

File: GameReview/GameReview/Controllers/UserController.cs [hunk 2]
```diff
@@ -1,70 +1,36 @@
...
-            }
-
-            // TODO what is this mapper
-            // map model to new user object
-            var user = _mapper.Map<User>(userModel);
+            UserDto userDto = new UserDto() 
+            { 
+                Username = userModel.Username,
+                Email = userModel.Email,
+                PasswordHash = userModel.Password
```
Comment (nisule) [inline GameReview/GameReview/Controllers/UserController.cs]: Think this should also be renamed to just Password  instead of PasswordHash

**Review by nisule — CHANGES_REQUESTED** [review]
Little renaming

File: GameReview/GameReview/Controllers/UserController.cs (hunk 2 above)
Comment (SamJDriver) [reply GameReview/GameReview/Controllers/UserController.cs]: renamed

File: GameReview/BusinessLogic/Infrastructure/UserService.cs (hunk 1 above)
Comment (SamJDriver) [reply GameReview/BusinessLogic/Infrastructure/UserService.cs]: renamed

**Review by nisule — APPROVED** [review]
LGTM

**Pull request merged**